aixplain/enums/function.py

```diff
@@ -27,9 +27,53 @@
 from aixplain.utils.request_utils import _request_with_retry
 from enum import Enum
 from urllib.parse import urljoin
+import logging
+import time
+from datetime import datetime, timedelta
+import json
+import os
+
+CACHE_FILE = ".aixplain_cache/functions.json"
+CACHE_DURATION = timedelta(hours=24)
 
+def save_to_cache(functions, functions_input_output):
+    def convert_sets(obj):
+        if isinstance(obj, dict): return {k: convert_sets(v) for k, v in obj.items()}
+        elif isinstance(obj, list): return [convert_sets(i) for i in obj]
+        elif isinstance(obj, set): return list(obj)
+        else: return obj
+    cache_data = {"timestamp": time.time(), "functions": {"enum": {name: value for name, value in functions.__members__.items()}, "input_output": convert_sets(functions_input_output)}}
+    temp_file = CACHE_FILE + ".tmp"
+    try:
+        os.makedirs(os.path.dirname(CACHE_FILE), exist_ok=True); json_str = json.dumps(cache_data); logging.info(f"Serialized JSON size: {len(json_str)} bytes")
+        with open(temp_file, "w") as f:
+            f.write(json_str); f.flush(); os.fsync(f.fileno())
+        os.replace(temp_file, CACHE_FILE); logging.info("Cache saved successfully.")
+    except Exception as e: logging.error(f"Failed to save cache: {e}")
 
+def convert_lists_to_sets(obj):
```

Comment:
Why do we need to do this conversion?

Reply:
I had a problem with parsing as I mentioned, I thought this might be related because JSON doesn’t support sets, so I changed the sets to lists when saving. I didn’t test it without this conversion after fixing the issue though. I will test removing the conversion, and if everything works fine without it, I’ll clean it up and remove this function.